submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/AccessTagProcessor.kt

```diff
@@ -1,14 +1,19 @@
 package ac.uk.ebi.biostd.submission.processors
 
 import ebi.ac.uk.model.ExtendedSubmission
+import ebi.ac.uk.model.extensions.releaseDate
 import ebi.ac.uk.persistence.PersistenceContext
 
 /**
  * Add all parent submission tags to submitted submission.
  */
 class AccessTagProcessor : SubmissionProcessor {
-
     override fun process(submission: ExtendedSubmission, context: PersistenceContext) {
-        submission.accessTags.addAll(context.getParentAccessTags(submission))
+        val accessTags =
+            context.getParentAccessTags(submission)
+                .toMutableList()
+                .apply { removeIf { it == "Public" && submission.releaseDate.isNullOrBlank().not() } }
+
+        submission.accessTags.addAll(accessTags)
```

Comment:
my suggestion was intended to something like 

```
submission.accessTags.addAll(accessTags).filter { it == "Public"}
```

and then in proper class
```
if (releaseDate > today){
     submission.addAccessTag("Public")
}
```

Reply:
Yes, I understand this, but maybe this is a mixing of concerns. Maybe this change must be made after the processors/handlers architecture is changed.

Otherwise, we'll be setting access tags in the time's processor or checking times in the access tags processor.

Reply:
Done